pkg/manifests/bundleloader.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(fileReader, 30)
+	csv := unstructured.Unstructured{}
+
+	if err = decoder.Decode(&csv); err != nil {
+		return nil
+	}
+
+	if csv.GetKind() != operatorsv1alpha1.ClusterServiceVersionKind {
+		return nil
+	}
+
+	var errs []error
+	bundle, err := loadBundle(csv.GetName(), filepath.Dir(path))
+	if err != nil {
+		errs = append(errs, fmt.Errorf("error loading objs in directory: %s", err))
+	}
+
+	if bundle == nil || bundle.CSV == nil {
+		errs = append(errs, fmt.Errorf("no bundle csv found"))
+		return utilerrors.NewAggregate(errs)
+	}
+
+	b.bundle = bundle
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// loadBundle takes the directory that a CSV is in and assumes the rest of the objects in that directory
+// are part of the bundle.
+func loadBundle(csvName string, dir string) (*Bundle, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	var errs []error
+	bundle := &Bundle{
+		Name: csvName,
+	}
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
+		if strings.HasPrefix(f.Name(), ".") {
+			continue
+		}
+
+		path := filepath.Join(dir, f.Name())
+		fileReader, err := os.Open(path)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("unable to load file %s: %s", path, err))
+			continue
+		}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(fileReader, 30)
+		obj := &unstructured.Unstructured{}
+		if err = decoder.Decode(obj); err != nil {
+			continue
+		}
+
+		switch kind := obj.GetKind(); kind {
+		case "ClusterServiceVersion":
+			if bundle.CSV != nil {
+				return nil, fmt.Errorf("invalid bundle: contains multiple CSVs")
+			}
+			csv := operatorsv1alpha1.ClusterServiceVersion{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+				obj.UnstructuredContent(),
+				&csv)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse CSV %s: %s", f.Name(), err.Error())
+			}
+			bundle.CSV = &csv
+		case "CustomResourceDefinition":
+			version := obj.GetAPIVersion()
+			if version == apiextensionsv1beta1.SchemeGroupVersion.String() {
+				crd := apiextensionsv1beta1.CustomResourceDefinition{}
+				err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+					obj.UnstructuredContent(),
+					&crd)
+				if err != nil {
+					return nil, fmt.Errorf("unable to parse CRD %s: %s", f.Name(), err.Error())
+				}
+				bundle.V1beta1CRDs = append(bundle.V1beta1CRDs, &crd)
+			} else if version == apiextensionsv1.SchemeGroupVersion.String() {
+				crd := apiextensionsv1.CustomResourceDefinition{}
+				err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+					obj.UnstructuredContent(),
+					&crd)
+				if err != nil {
+					return nil, fmt.Errorf("unable to parse CRD %s: %s", f.Name(), err.Error())
+				}
+				bundle.V1CRDs = append(bundle.V1CRDs, &crd)
+			} else {
+				return nil, fmt.Errorf("unsupported CRD version %s for %s", version, f.Name())
+			}
+		default:
+			bundle.Objects = append(bundle.Objects, obj)
```

Comment:
One difference I found between this implementation and that in OR: all objects in a bundle, including CSVs and CRDs, are [appended](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/directory.go#L224) to `Bundle.Objects`. Is it reasonable to have the same behavior here?

Reply:
Ah, yes, it is totally reasonable and the correct behavior. I'll push a change to this pr